src/sagas/CreateClaim.saga.test.ts

```diff
@@ -0,0 +1,179 @@
+  
+  assert({
+    given: 'Work Claim Form Submit Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM, handleOnCreateClaim),
+  })
+
+  assert({
+    given: 'Work Claim Form Submit Success Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM_SUCCESS, handleOnCreateClaimSuccess),
+  })
+
+  assert({
+    given: 'Work Claim Form Submit Error Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM_ERROR, handleOnCreateClaimError),
+  })
+})
+
+describe('handleOnCreateClaim()', async assert => {
+  const token = 'test-token'
+  const work = createWork()
+  const workId = 'testWorkId'
+
+  const iterator = handleOnCreateClaim(Actions.CreateClaim.onCreateClaim({ work, token }))
+
+  assert({
+    given: 'Work Claim Form Submit Action',
+    should: 'set loading status',
+    actual: iterator.next().value,
+    expected: put(Actions.LoadingPage.onLoadingOn()),
+  })
+
+  {
+    const frost = new Frost({ host: Configuration.frostApiUrl })
+    assert({
+      given: 'next step',
+      should: 'post work with frost',
+      actual: iterator.next().value,
+      expected: call([frost, frost.createWork], token, work),
+    })
+  }
+
+  assert({
+    given: 'next step',
+    should: 'clear loading',
+    actual: iterator.next({ workId }).value,
+    expected: put(Actions.LoadingPage.onLoadingFull())
+  })
+
+  assert({
+    given: 'next step',
+    should: 'reset notifaction bar',
+    actual: iterator.next().value,
+    expected: put(Actions.CreateClaim.onCreateClaimSuccess({ workId })),
+  })
+})
+
+describe('handleOnCreateClaimSuccess', async assert => {
+  const workId = 'testWorkId'
+
+  const iterator = handleOnCreateClaimSuccess(Actions.CreateClaim.onCreateClaimSuccess({ workId }))
+
+  assert({
+    given: 'Work Claim Form Submit Success Action',
+    should: 'set workId in notification bar',
+    actual: iterator.next().value,
+    expected: put(Actions.NotificationBar.onShowNotificationBar({
+      type: 'link-success',
+      message: `${Configuration.mainExplorerUrl}/works/${workId}`,
+    })),
+  })
+
+  assert({
+    given: 'next step',
+    should: 'delay 3000',
+    actual: iterator.next().value,
+    expected: call(delay, 3000),
+  })
+
+  assert({
+    given: 'next step',
+    should: 'hide notifaction bar',
+    actual: iterator.next().value,
+    expected: put(Actions.NotificationBar.onHideNotificationBar()),
+  })
+
+  assert({
+    given: 'next step',
+    should: 'delay 2000',
+    actual: iterator.next().value,
+    expected: call(delay, 2000),
+  })
+
+  assert({
+    given: 'next step',
+    should: 'reset notifaction bar',
+    actual: iterator.next().value,
+    expected: put(Actions.NotificationBar.onResetNotificationBar()),
```

Comment:
I am not familiar with the notification bar, this is probably out of the scope of this pr, but why do we need to reset the notification bar? what if some other notification is being displayed and we reset the notification bar does it also wipe out that other notification?

Reply:
The hide action before this one just sets the classname to start the transition to opacity of 0. The reset action sets the css property `display: none`. 

Yeah if there is another notification that is displayed right before this resets(or hides) then it will reset(or hide) that notification as well. 

For example if you create an apiToken and then immediately create the claim with final form, the createClaim success notification will show in place of the createToken success notification, but whenever the createToken success notification hides/resets it will do the same to the createClaim success notification.

Reply:
Ok thanks, out of scope of this PR, but that is something that should be fixed, maybe not in this repo thought sense we are planning on rebuilding this app from scratch.